components/dashboard/src/admin/TeamDetail.tsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import moment from "moment";
+import { useState } from "react";
+import { Team, TeamMemberInfo, TeamMemberRole } from "@gitpod/gitpod-protocol";
+import { getGitpodService } from "../service/service";
+import { Item, ItemField, ItemsList } from "../components/ItemsList";
+import DropDown from "../components/DropDown";
+import { Link } from "react-router-dom";
+import Label from "./Label";
+
+export default function TeamDetail(props: { team: Team, members: TeamMemberInfo[] }) {
+    const { team, members } = props;
+    const [teamMembers, setTeamMembers] = useState<TeamMemberInfo[] | undefined>(members);
+    const [searchText, setSearchText] = useState<string>('');
+
+    const filteredMembers = teamMembers?.filter(m => {
+        const memberSearchText = `${m.fullName || ''}${m.primaryEmail || ''}`.toLocaleLowerCase();
+        if (!memberSearchText.includes(searchText.toLocaleLowerCase())) {
+            return false;
+        }
+        return true;
+    });
+
+    const setTeamMemberRole = async (userId: string, role: TeamMemberRole) => {
+        await getGitpodService().server.adminSetTeamMemberRole(team!.id, userId, role);
+        setTeamMembers(await getGitpodService().server.adminGetTeamMembers(team!.id));
+    }
+    return <>
+        <div className="flex">
+            <div className="flex-1">
```

Comment:
**issue:** Just run into an issue that affects both Projects and Teams in the admin dashboard. Certainly out of the scope of this PR, but when you reload the details page for a project or a team, parts of the information in the header disappears and does not load. This behaves also slightly different for deleted teams. Could you open a follow up issue for this if it's reproducible?

https://user-images.githubusercontent.com/120486/155565249-f8333bec-e5a1-47c5-aa90-ee9178d83c42.mov

Reply:
This confused me too - I checked on prod and it's not reproducible there. I wonder if there are websocket connection limitations when it comes to preview envs 🤔

Reply:
Also not reproducible on staging. I'll make a note to check this on staging after merging, and if it's reproducible, I'll create a follow-up issue 🙏🏽